src/main/test/java/io/burt/jmespath/contrib/function/JmesPathRuntimeWithStringFunctionTest.java

```diff
@@ -0,0 +1,586 @@
+    search("concat(@)", parse("{}"));
+  }
+
+  @Test
+  public void lowerCaseExampleFromXPathSpec() {
+    T result = search("lower_case('ABc!D')", dontCare);
+    assertThat(result, is(jsonString("abc!d")));
+  }
+
+  @Test
+  public void lowerCaseRequiresASingleArgument() {
+    thrown.expect(ParseException.class);
+    thrown.expectMessage(containsString("invalid arity calling \"lower_case\" (expected 1 but was 2)"));
+    search("lower_case(@, @)", dontCare);
+  }
+
+  @Test
+  public void lowerCaseRequiresAStringAsArgument() {
+    thrown.expect(ArgumentTypeException.class);
+    thrown.expectMessage(containsString("expected string but was array"));
+    search("lower_case(@)", parse("[3]"));
+  }
+
+  @Test
+  public void upperCaseExamplesFromXPathSpec() {
+    T result = search("upper_case('abCd0')", dontCare);
+    assertThat(result, is(jsonString("ABCD0")));
+  }
+
+  @Test
+  public void upperCaseRequiresASingleArgument() {
+    thrown.expect( ParseException.class);
+    thrown.expectMessage(containsString("invalid arity calling \"upper_case\" (expected 1 but was 2)"));
+    search("upper_case(@, @)", dontCare);
+  }
+
+  @Test
+  public void upperCaseRequiresAStringAsArgument() {
+    thrown.expect(ArgumentTypeException.class);
+    thrown.expectMessage(containsString("expected string but was object"));
+    search("upper_case(@)", dontCare);
+  }
+
+  @Test
+  public void normalizeSpaceExamplesFromXPathSpec() {
+    T result = search("normalize_space(@)", parse("\" The    wealthy curled darlings\\n" +
+            "                             of    our    nation. \""));
+    assertThat(result, is(jsonString("The wealthy curled darlings of our nation.")));
+  }
+
+  @Test
+  public void normalizeSpaceRemovesLeadingWhitespaces() {
+    T result = search("normalize_space(str)", parse("{ \"str\" : \"\\n\\t \\tend\"}"));
+    assertThat(result, is(jsonString("end")));
+  }
+
+  @Test
+  public void normalizeSpaceRemovesTrailingWhitespaces() {
+    T result = search("normalize_space(str)", parse("{ \"str\" : \"begin\\n \\t \\t\"}"));
+    assertThat(result, is(jsonString("begin")));
+  }
+
+  @Test
+  public void normalizeSpaceCollapseInnerWhitespaces() {
+    T result = search("normalize_space(str)", parse("{ \"str\" : \"begin\\n\\t \\tend\"}"));
+    assertThat(result, is(jsonString("begin end")));
+  }
+
+  @Test
+  public void normalizeSpaceRequiresASingleArgument() {
+    thrown.expect( ParseException.class );
+    thrown.expectMessage(containsString("invalid arity calling \"normalize_space\" (expected 1 but was 2)"));
+    search("normalize_space(@, @)", dontCare);
+  }
+
+  @Test
+  public void normalizeSpaceRequiresAStringAsArgument() {
+    thrown.expect(ArgumentTypeException.class);
+    thrown.expectMessage(containsString("expected string but was object"));
+    search("normalize_space(@)", dontCare);
+  }
+
+  @Test
+  public void translateExamplesFromXpathSpec() {
```

Comment:
This doesn't explain what translate does, and there doesn't seem to be any other tests that overlap. I think there should be three tests for these three examples, each with a descriptive name – like above with the whitespace normalization.